dsl/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/dsl/jpql/Jpql.kt

```diff
@@ -646,6 +646,70 @@ open class Jpql : JpqlDsl {
         return Expressions.ln(value.toExpression())
     }
 
+    /**
+     * Creates an expression that represents the mod of values.
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any> mod(expr: KProperty1<T, Int>, value: Int): Expression<Int> {
+        return Expressions.mod(Paths.path(expr), Expressions.value(value))
+    }
+
+    /**
+     * Creates an expression that represents the mod of values.
+     */
+    @SinceJdsl("3.4.0")
+    fun mod(value1: Expressionable<Int>, value2: Int): Expression<Int> {
+        return Expressions.mod(value1.toExpression(), Expressions.value(value2))
+    }
+
+    /**
+     * Creates an expression that represents the mod of values.
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any> mod(expr: KProperty1<T, Int>, value: Expression<Int>): Expression<Int> {
+        return Expressions.mod(Paths.path(expr), value.toExpression())
+    }
+
+    /**
+     * Creates an expression that represents the mod of values.
+     */
+    @SinceJdsl("3.4.0")
+    fun mod(value1: Expressionable<Int>, value2: Expressionable<Int>): Expression<Int> {
+        return Expressions.mod(value1.toExpression(), value2.toExpression())
+    }
+
+    /**
+     * Create expression that calculates the powering of a numeric [base] to a specified [exponent].
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Any, V : Number> power(base: KProperty1<T, @Exact V>, exponent: Int): Expression<V> {
+        return Expressions.power(Paths.path(base), Expressions.value(exponent))
+    }
+
+    /**
+     * Create expression that calculates the powering of a numeric [base] to a specified [exponent].
+     */
+    @SinceJdsl("3.4.0")
+    fun <T : Number> power(base: Expressionable<T>, exponent: Int): Expression<T> {
```

Comment:
<img width="629" alt="powerfunction" src="https://github.com/line/kotlin-jdsl/assets/13346446/3b65b6ee-9005-46b8-aace-28bb4773ddc0">

Could you please write a method based on the specification in [jakarta specification](https://jakarta.ee/specifications/persistence/3.1/jakarta-persistence-spec-3.1.html#a5284)?

Reply:
Wow, I love doing reviews based on such solid knowledge. lol Thank you :)

I feel like this allows me to get up close and personal with the Hibernate spec.

Reply:
Oh... Thank you.
I gained a lot of knowledge while implementing this `power`.
I also understand now why the package was changed from javax to jakrta.

References
- https://medium.com/@nagarjunmallesh/java-persistence-api-vs-jakarta-persistence-layer-a-comparative-study-with-code-implementation-4462b1abcfd1
- https://www.samsungsds.com/kr/insights/java_jakarta.html